src/logging/utils.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { asError } from '@/logging/utils';
+
+describe('asError', () => {
+  it('should return the same error if thrown is an instance of Error', () => {
+    const thrown = new Error('test error');
+
+    expect(asError(thrown)).toEqual(new Error('test error'));
+  });
+
+  it('should return the same error if thrown is a superset of Error', () => {
+    class CustomError extends Error {
+      constructor(message: string) {
+        super(message);
+      }
+    }
+    const thrown = new CustomError('test error');
+
+    expect(asError(thrown)).toEqual(new CustomError('test error'));
+  });
+
+  it('should return a new Error instance with the thrown value if thrown is a string', () => {
+    const thrown = 'test error';
+
+    const result = asError(thrown);
+    expect(result).toEqual(new Error('test error'));
+
+    // If stringified:
+    expect(result).not.toEqual(new Error('"test error'));
+  });
+
+  it('should return a new Error instance with the message thrown value if thrown is object with message', () => {
+    const thrown = { message: 'test error' };
+
+    const result = asError(thrown);
+    expect(result).toEqual(new Error('test error'));
+
+    // If stringified:
+    expect(result).not.toEqual(new Error('{"message":"test error"}'));
```

Comment:
What is the reason to test the equality in one assertion and testing a different one (non-equality) in the following assertion?

Reply:
I wanted to make sure that the `message` was used and not the object.